src/VisualStudio/Core/Def/RoslynDocumentProvider.cs

```diff
@@ -40,11 +40,15 @@ protected override void OnBeforeNonRoslynDocumentWindowShow(IVsWindowFrame frame
                 return;
             }
 
-            var view = GetTextViewFromFrame(frame);
-            if (view != null)
+            // Schedule hookup of the non-Roslyn document view on foreground task scheduler.
+            InvokeBelowInputPriority(() =>
```

Comment:
Ah, I see that will end up executing right away as we are on foreground thread already. I'll change this to explicitly invoke on a scheduled task.